modules/ROOT/pages/sdk-xattr-example.adoc

```diff
@@ -186,3 +186,62 @@ public class CouchbaseExtendedAttributesDemo
     }
 }
 ----
+
+== Virtual Extended Attributes Example
+
+Using the subdocument API, xref:sdk-xattr-overview.adoc#_virtual_extended_attributes[Virtual XATTR] can be used to fetch metadata about a document, via the `$document` virtual XATTR.
+A common use case is discovering documentation expiry data, or TTL:
+
+[source,java]
+----
+import java.text.SimpleDateFormat;
+import java.util.Date;
+ 
+import com.couchbase.client.core.message.kv.subdoc.multi.Lookup;
+import com.couchbase.client.java.Bucket;
+import com.couchbase.client.java.Cluster;
+import com.couchbase.client.java.CouchbaseCluster;
+import com.couchbase.client.java.subdoc.*;
+ 
+public class GetTTL
+{
+    public static void main(String [] args)
+    {
+        // Access the cluster that is running on the local host, authenticating with
+        // the username and password of the Full Administrator. This
+        // provides all privileges.
+        //
+        Cluster cluster = CouchbaseCluster.create("localhost");
+        cluster.authenticate("test", "password");
+        Bucket test = cluster.openBucket("test");
+ 
+        String key = "Perry";
+ 
+        DocumentFragment<Lookup> click = test.lookupIn(key).get("click").execute();
+        System.out.println("Click: " + click.content("click"));
+ 
+        DocumentFragment<Lookup> ttl = test.lookupIn(key).get("$document.exptime", new SubdocOptionsBuilder().xattr(true)).execute();
+        System.out.println("ttl: " + ttl.content("$document.exptime"));
+ 
+        test.touch(key, 300);
+ 
+        ttl = test.lookupIn(key).get("$document.exptime", new SubdocOptionsBuilder().xattr(true)).execute();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy/MM/dd HH:mm.ss");
+        System.out.println("ttl: " + sdf.format(new Date(ttl.content("$document.exptime", int.class))));
+ 
+        test.touch(key, 0);
+ 
+        //Multiple paths can be access at once via subdoc. It's limited to 16 paths and xattrs have to be frist
+        DocumentFragment<Lookup> mutlisubpath = test.lookupIn(key).get("$document.exptime", new SubdocOptionsBuilder().xattr(true))
```

Comment:
What's a "mutlisubpath"?  It's both misspelled and not camel case.  :(

Reply:
looks like you're still working on this one I believe.

Reply:
Yes - code sample still to finish changing.